### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by fisx**
- some haddocks, formatting
- re-arrange helpers
- make API client functions in integration tests more polymorphic (needed for upcoming tests)
- none of these changes should change behavior
- i recomment reading commit by commit

### CODE REVIEW

**Review by jschaul — APPROVED** [review]

File: services/galley/src/Galley/API/LegalHold.hs [hunk 1]
```diff
@@ -204,6 +204,13 @@ requestDevice zusr tid uid = do
     Just UserLegalHoldDisabled -> RequestDeviceSuccess <$ provisionLHDevice
     Nothing -> throwM teamMemberNotFound
   where
+    -- LH service should be idempotent in device creation, ie. return the existing device on
```
[the remark this answers is not in the document]
Comment (fisx) [reply services/galley/src/Galley/API/LegalHold.hs]: 453c12a

**Comment by jschaul** [comment]
Failing test:

```
      GET, PUT [/i]?/teams/{tid}/legalhold - too large:

FAIL

        Exception: No code?

        CallStack (from HasCallStack):

          error, called at test/integration/API/Util.hs:404:40 in main:API.Util

          getInvitationCode, called at test/integration/API/Util.hs:337:23 in main:API.Util

          addUserToTeamWithRole', called at test/integration/API/Util.hs:320:18 in main:API.Util

          addUserToTeamWithRole, called at test/integration/API/Util.hs:313:17 in main:API.Util

          addUserToTeam, called at test/integration/API/Util.hs:109:12 in main:API.Util

          createBindingTeamWithMembers, called at test/integration/API/Teams/LegalHold.hs:536:29 in main:API.Teams.LegalHold

```

**Comment by fisx** [comment]
Pretty sure this is unrelated, but I've made an attempt at fixing it under the assumption it's a race condition: c331dea

**Comment by fisx** [comment]
(couldn't reproduce locally, btw)

**Pull request merged**